## HW 3 Assignment III
###  Q: is the new Bus route improving commute?

### H0: the commute time is the same or longer with the new bus route as it was before
### Tnew >= Told 

### Ha: the commute time is shorter with the new bus route as it was before with a significance threshold of a 5%
### Tnew < Told, α = 0.05

### Know information about the previous route (population) N(μ=36, σ=6)

In [40]:
from __future__ import print_function
import pandas as pd
import numpy as np
import math

#Obtain the time table from fb55 repo using the curl command and storing a locla copy
!curl -O https://raw.githubusercontent.com/fedhere/PUI2016_fb55/master/Lab3_fb55/times.txt

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
107  1179  107  1179    0     0   9013      0 --:--:-- --:--:-- --:--:--  575k


In [38]:
#Load the timearray into a numpy array, caculate mean and standard deviation
timearray = np.loadtxt('./times.txt', dtype=float)
print ("New route mean: " + str(timearray.mean()))
print ("New route std:   " + str(timearray.std()))
print ("The lenght of the sample is: " + str(len(timearray)))

#Calculating the z-test

#Z ~ (mean(pop) - mean(sample)) / (stdev / sqrt(N))

New route mean: 34.4661616883
New route std:   7.10150406819
The lenght of the sample is: 100


The Z-Test is calculated as

$z=\frac{\bar{x}-\mu}{\sigma/\sqrt{n}}$

In [48]:
#From the original problem information we know that μ=36, σ=6)
z = (34.47 - 36)/ (6/math.sqrt(100))
print ("The Z-score is: " + str(z))

The Z-score is: -2.55


That z scores means that the mean time travel on the new route is 2.55 standard deviations away from the previous, which is greater than the two standard deviations (5%), which enables us to falsify our hypotesis